### PR TITLE
Support local train the Keras model defined in ElasticDL.

### PR DESCRIPTION
**Pull request by workingloong**
Implement #1578
Implement a local executor which can train the Kears model in model zoo using custom training loop with Tensorflow low-level API.

### CODE REVIEW

File: scripts/client_test.sh [hunk 1]
```diff
@@ -100,6 +100,18 @@ elif [[ "$JOB_TYPE" == "odps" ]]; then
       --log_level=INFO \
       --image_pull_policy=Never \
       --output=model_output
+elif [[ "$JOB_TYPE" == "train" ]]; then
```
Comment (terrytangyuan) [inline scripts/client_test.sh]: train -> local

File: elasticdl/python/common/args.py [hunk 2]
```diff
@@ -510,13 +510,14 @@ def add_common_args_between_master_and_worker(parser):
...
             "",
+            DistributionStrategy.LOCAL,
             DistributionStrategy.PARAMETER_SERVER,
             DistributionStrategy.ALLREDUCE,
         ],
         default="",
         help="Master will use a distribution policy on a list of devices "
         "according to the distributed strategy, "
-        'e.g. "ParameterServerStrategy" or "AllreduceStrategy"',
+        'e.g. "ParameterServerStrategy" or "AllreduceStrategy" or "local"',
```
Comment (brightcoder01) [inline elasticdl/python/common/args.py]: local -> Local

File: elasticdl/python/elasticdl/api.py [hunk 3]
```diff
@@ -5,45 +5,52 @@
...
 from elasticdl.python.elasticdl.image_builder import (
     build_and_push_docker_image,
     remove_images,
 )
+from elasticdl.python.elasticdl.local_executor import LocalExecutor
 
 
 def train(args):
     model_zoo = os.path.normpath(args.model_zoo)
+    print(args.distribution_strategy)
```
Comment (brightcoder01) [inline elasticdl/python/elasticdl/api.py]: This line is just for debug, right?

Comment (brightcoder01) [inline elasticdl/python/elasticdl/api.py]: We can remove this line of code for debugging.

File: elasticdl/python/tests/local_executor_test.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            )
+
+            train_dataset = local_executor._get_dataset(train_tasks)
+            for features, labels in train_dataset.take(1):
+                loss = local_executor._train(features, labels)
+                self.assertEqual(type(loss.numpy()), np.float32)
+
+            validation_dataset = local_executor._get_dataset(validation_tasks)
+            metrics = local_executor._evaluate(validation_dataset)
+            self.assertEqual(list(metrics.keys()), ["accuracy"])
```
Comment (skydoorkai) [inline elasticdl/python/tests/local_executor_test.py:112]: Why add these two ` assertEqual` to test type and dict key names?

File: elasticdl/python/elasticdl/local_executor.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+        train_tasks = self._gen_tasks(self.training_data)
+        validation_tasks = self._gen_tasks(self.validation_data)
+        train_dataset = self._get_dataset(train_tasks)
+        validation_dataset = self._get_dataset(validation_tasks)
+
+        while epoch < self.epoch:
+            for features, labels in train_dataset:
+                loss = self._train(features, labels)
+                logger.info("Loss = {}".format(loss))
```
Comment (skydoorkai) [inline elasticdl/python/elasticdl/local_executor.py]: Add `step` in logger for `Loss`, so the user can see the progress more clearly.

File: elasticdl/python/tests/local_executor_test.py (hunk 4 above)
Comment (workingloong) [reply elasticdl/python/tests/local_executor_test.py:112]: I just want to test `evaluate` can return metrics.

File: elasticdl/python/elasticdl/local_executor.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+                ):
+                    self._evaluate(validation_dataset)
+            self._evaluate(validation_dataset)
+            logger.info("Epoch {} end".format(epoch))
+            epoch += 1
+
+    def _train(self, features, labels):
+        with tf.GradientTape() as tape:
+            outputs = self.model_inst.call(features, training=True)
+            loss = self.loss_fn(labels, outputs)
```
Comment (skydoorkai) [inline elasticdl/python/elasticdl/local_executor.py:100]: Add regularization loss if it exists.
See [worker.py](https://github.com/sql-machine-learning/elasticdl/blob/645a1399efad2629d0a356ee3f5409dbfae3c6ea/elasticdl/python/worker/worker.py#L643)

**Review by skydoorkai — APPROVED** [review]

**Comment by workingloong** [comment]
Fix #1578
